### PR TITLE
DM-40225: Update to use the new non-tenant user for S3 bucket access

### PR DESCRIPTION
**Pull request by hsinfang**

### CODE REVIEW

File: kubernetes/overlays/dev/prompt-proto-service/prompt-proto-service.yaml [hunk 1]
```diff
@@ -30,13 +30,15 @@ spec:
...
+          value: rubin:rubin-pp
         - name: BUCKET_TOPIC
           value: rubin-prompt-processing
         - name: IMAGE_TIMEOUT
           value: '120'
         - name: CALIB_REPO
-          value: s3://rubin-pp-users/central_repo/
+          value: s3://rubin:rubin-pp-users/central_repo/
+        - name: LSST_DISABLE_BUCKET_VALIDATION
+          value: '1'
```
Comment (kfindeisen) [inline kubernetes/overlays/dev/prompt-proto-service/prompt-proto-service.yaml:41]: I don't understand Boto well enough to review all this, but I have one question. Does this variable need to go in [the Playbook](https://github.com/lsst-dm/prompt_prototype/blob/main/doc/playbook.rst#usdf-1)? I don't see it used in `activator.py`, so I'm not sure whether that's the right context.

Comment (hsinfang) [reply kubernetes/overlays/dev/prompt-proto-service/prompt-proto-service.yaml:41]: This env var needs to be there at Butler instantiation so Butler doesn't complain about the bucket name. Good point I should update the Playbook. Thank you for pointing it out! 

For reference, the traceback without this env var is the following: 
```
  File "/app/activator/activator.py", line 97, in <module>
    central_butler = get_central_butler(calib_repo, instrument_name)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/activator/middleware_interface.py", line 70, in get_central_butler
    registry = Butler(central_repo).registry
               ^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-7.0.1/Linux64/daf_butler/ga997e9e321+005860e1e6/python/lsst/daf/butler/_butler.py", line 225, in __init__
    self._config = ButlerConfig(config, searchPaths=searchPaths, without_datastore=without_datastore)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-7.0.1/Linux64/daf_butler/ga997e9e321+005860e1e6/python/lsst/daf/butler/_butlerConfig.py", line 127, in __init__
    butlerConfig = Config(other)
                   ^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-7.0.1/Linux64/daf_butler/ga997e9e321+005860e1e6/python/lsst/daf/butler/core/config.py", line 275, in __init__
    self.__initFromUri(other)
  File "/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-7.0.1/Linux64/daf_butler/ga997e9e321+005860e1e6/python/lsst/daf/butler/core/config.py", line 364, in __initFromUri
    content = uri.read()
              ^^^^^^^^^^
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/backoff/_sync.py", line 105, in retry
    ret = target(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-7.0.1/Linux64/resources/g9349d46f9b+a4f294311b/python/lsst/resources/s3.py", line 154, in read
    response = self.client.get_object(Bucket=self.netloc, Key=self.relativeToPathRoot, **args)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/botocore/client.py", line 534, in _api_call
    return self._make_api_call(operation_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/botocore/client.py", line 927, in _make_api_call
    api_params = self._emit_api_params(
                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/botocore/client.py", line 1039, in _emit_api_params
    self.meta.events.emit(
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/botocore/hooks.py", line 412, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/botocore/hooks.py", line 256, in emit
    return self._emit(event_name, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/botocore/hooks.py", line 239, in _emit
    response = handler(**kwargs)
               ^^^^^^^^^^^^^^^^^
  File "/opt/lsst/software/stack/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-7.0.1/lib/python3.11/site-packages/botocore/handlers.py", line 284, in validate_bucket_name
    raise ParamValidationError(report=error_msg)
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid bucket name "rubin:rubin-pp-users": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN matching the regex "^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
```

**Review by dspeck1 — APPROVED** [review]
Looks good

File: kubernetes/overlays/dev/prompt-proto-service/prompt-proto-service.yaml (hunk 1 above)
Comment (hsinfang) [reply kubernetes/overlays/dev/prompt-proto-service/prompt-proto-service.yaml:41]: Playbook is updated  https://github.com/lsst-dm/prompt_prototype/pull/76/commits/d168d1ddbe53e879c6fe3ea6f7d6f2f04bcb3c14

**Pull request merged**